app/javascript/components/set-ownership-form/index.jsx

```diff
@@ -8,26 +8,50 @@ import createSchema from './ownership-form.schema';
 class SetOwnershipForm extends Component {
   state = {
     initialValues: {},
-  }
+    ownerOptions: [],
+    groupOptions: [],
+  };
 
   componentDidMount() {
     const { ownershipIds } = this.props;
     this.loadInitialData(ownershipIds);
   }
 
-  loadInitialData = objectIds =>
-    http.post(`/${ManageIQ.controller}/ownership_form_fields`, { object_ids: objectIds })
-      .then(data => this.setState({ initialValues: data }))
+  loadInitialData = (ownershipIds) => {
+    Promise.all([
+      /**
+       * Loads all the users with id and name into the ownerOptions state.
+       */
+      http.get('/api/users?expand=resources'),
+
+      /**
+       * Loads all the groups with id and description into the groupOptions state.
+       */
+      http.get('/api/groups?expand=resources'),
 
+      /**
+       * Loads the values which are set for the current object(s).
+       * None set results in: {user: null, group: 1}
+       * Multiple selected results in: {???}
+       */
+      http.post(`/${ManageIQ.controller}/ownership_form_fields`, { object_ids: ownershipIds }),
+    ])
+      .then(data =>
```

Comment:
If you destructure this into 3 separate variables, it's a little more readable and requires no indexing below.